src/blob/index/core.rs

```diff
@@ -59,14 +59,35 @@ struct MemoryAttrs<K> {
     marker: PhantomData<K>,
 }
 
+const BTREE_B_FACTOR: usize = 6;
+const BTREE_VALUES_LEN: usize = BTREE_B_FACTOR * 2 - 1;
+const BTREE_EDGES_LEN: usize = BTREE_B_FACTOR * 2;
+
 impl<K> MemoryAttrs<K>
 where
     for<'a> K: Key<'a>,
 {
     const KEY_SIZE: usize = K::LEN as usize;
-    const BPTREE_ENTRY_SIZE: usize =
+    const BTREE_ENTRY_SIZE: usize =
         size_of::<Vec<u8>>() + MemoryAttrs::<K>::KEY_SIZE + size_of::<Vec<RecordHeader>>();
     const RECORD_HEADER_SIZE: usize = size_of::<RecordHeader>() + MemoryAttrs::<K>::KEY_SIZE;
+    // Each node in BTreeMap contains preallocated vectors of 11 values and 12 edges.
+    // Although count of nodes can't be determined without reimplementing insertion algorithm,
+    // we can use approximation of overhead size added per one key
+    const BTREE_DATA_NODE_SIZE: usize = size_of::<Option<std::ptr::NonNull<()>>>() +                     // ptr to parent
+                                        size_of::<u16>() * 2 +                                           // metadata
+                                        MemoryAttrs::<K>::BTREE_ENTRY_SIZE * BTREE_VALUES_LEN;           // data
+    const BTREE_DATA_NODE_RATIO: f64 = 1.0 / BTREE_VALUES_LEN as f64;
+    const BTREE_INTERNAL_NODE_OVERHEAD: usize = size_of::<std::ptr::NonNull<()>>() * BTREE_EDGES_LEN;    // edges
+    const BTREE_INTERNAL_NODE_RATIO: f64 = (((1.0 +
+                                              BTREE_EDGES_LEN as f64 +
+                                              BTREE_EDGES_LEN.pow(2) as f64 +
+                                              BTREE_EDGES_LEN.pow(3) as f64 +
+                                              BTREE_EDGES_LEN.pow(4) as f64)
+                                             / (BTREE_VALUES_LEN as f64 * BTREE_EDGES_LEN.pow(5) as f64)));
```

Comment:
You can do the type conversion once for the numerator and once for the denominator